clients/js/algorand.ts

```diff
@@ -0,0 +1,164 @@
+  }
+  let contracts = CONTRACTS[environment][chainName];
+
+  let target_contract: string;
+
+  switch (payload.module) {
+    case "Core":
+      target_contract = contracts.core;
+      // sigh...
+      switch (payload.type) {
+        case "GuardianSetUpgrade":
+          console.log("Submitting new guardian set");
+          break;
+        case "ContractUpgrade":
+          console.log("Upgrading core contract");
+          break;
+        default:
+          impossible(payload);
+      }
+      break;
+    case "NFTBridge":
+      if (contracts.nft_bridge === undefined) {
+        // NOTE: this code can safely be removed once the terra NFT bridge is
+        // released, but it's fine for it to stay, as the condition will just be
+        // skipped once 'contracts.nft_bridge' is defined
+        throw new Error("NFT bridge not supported yet for terra");
+      }
+      target_contract = contracts.nft_bridge;
+      switch (payload.type) {
+        case "ContractUpgrade":
+          console.log("Upgrading contract");
+          break;
+        case "RegisterChain":
+          console.log("Registering chain");
+          break;
+        case "Transfer":
+          console.log("Completing transfer");
+          break;
+        default:
+          impossible(payload);
+      }
+      break;
+    case "TokenBridge":
+      if (contracts.token_bridge === undefined) {
+        throw new Error("contracts.token_bridge is undefined");
+      }
+      target_contract = contracts.token_bridge;
+      switch (payload.type) {
+        case "ContractUpgrade":
+          console.log("Upgrading contract");
+          break;
+        case "RegisterChain":
+          console.log("Registering chain");
+          break;
+        case "Transfer":
+          console.log("Completing transfer");
+          break;
+        case "AttestMeta":
+          console.log("Creating wrapped token");
+          break;
+        case "TransferWithPayload":
+          throw Error("Can't complete payload 3 transfer from CLI");
+        default:
+          impossible(payload);
+          break;
+      }
+      break;
+    default:
+      target_contract = impossible(payload);
+  }
+
+  const CORE_ID = BigInt(parseInt(contracts.core));
+  const TOKEN_BRIDGE_ID = BigInt(parseInt(contracts.token_bridge));
+
+  const algodClient = new Algodv2(
+    ALGORAND_HOST.algodToken,
+    ALGORAND_HOST.algodServer,
+    ALGORAND_HOST.algodPort
+  );
+  const algoWallet: Account = mnemonicToSecretKey(n.key);
+
+  if (await getIsTransferCompletedAlgorand(algodClient, TOKEN_BRIDGE_ID, vaa)) {
+    console.log("Vaa has already been redeemed.");
+    return;
+  }
+
+  // Create transaction
+  const txs = await redeemOnAlgorand(
+    algodClient,
+    TOKEN_BRIDGE_ID,
+    CORE_ID,
+    vaa,
+    algoWallet.addr
+  );
+  // Sign and send transaction
+  const result = await signSendAndConfirmAlgorand(algodClient, txs, algoWallet);
+}
```

Comment:
You're not logging anything? If it's success, can you log the transaction hash? Otherwise just dump result?

Reply:
Logging confirming block